SCClassLibrary/Common/Collections/Scale.sc

```diff
@@ -185,8 +193,17 @@ Tuning {
 	}
 
 	*doesNotUnderstand { |selector, args|
-		var tuning = this.newFromKey(selector, args).deepCopy;
-		^tuning ?? { super.doesNotUnderstand(selector, args) }
+		^this.newFromKey(selector, args).deepCopy
+		!? { |v| v } //return if valid
+		?? { super.doesNotUnderstand(selector, args) }
+	}
+
+	*doesNotUnderstandWithKeys {|selector, argsArray, keywordArgsAsPairs|
+		^Tuning.performWithEnvir(
+			\newFromKey,
+			Tuning.class.findRespondingMethodFor(\newFromKey)
+			.makePerformableEnvir([selector] ++ argsArray, keywordArgsAsPairs)
+		)
```

Comment:
If `makePerformableEnvir` were implemented as method of Object, this code would look a bit simpler:

```
^Tuning.performWithEnvir(
			\newFromKey,
			Tuning.class.makePerformableEnvir([selector] ++ argsArray, keywordArgsAsPairs)
		)
```

Actually I am not sure I understand why the Method `newFromKey` is used to build a performableEnvir for a different `selector`. Couldn't this be the Method found via the `selector`?

Reply:
It's `Scale.major(tuning: Tuning.choose)` so `\major` is the selector and you still need to build the arguments needed by `\newFromKey`.... I used scale here because I can't think of the name of a tuning, but it's the same.

I don't think it is necessary to add this to object because it would still have to go through Method/Functiondef to actually get the arguments. Also object is huge.

Reply:
I see what you mean. 

Here is another suggestion. When I wrote `performWithEnvir`, I didn't think of passing non-keyword arguments. We could now extend this method to take an array of arguments.

Then the functionality of `makePerformableEnvir` could go in there. Some of the functionality is already there, and duplicating it in an extra method is perhaps not optimal anyway.

Then, in the above case, you could simply call 
```
Tuning.performWithEnvir(\newFromKey, keywordArgsAsPairs.asDict, argsArray)
```

For Function, this would be `valueWithEnvir`, which would take a number of arguments as a second argument.

**BUT/AND**: Of course, then, it may also be a simplification to keep the method `makePerformableArray` that converts the mixed data from array and envir to one single array that can be used to call `valueArray` and `performList` within the `valueWithEnvir` and `performWithEnvir` methods.

**SO:** `makePerformableEnvir`  may be dropped, `valueWithEnvir` and `performWithEnvir` improved by using `makePerformableArray`, and adding an extra array argument. All this is without too much extra code.